figures/pipeline/enrollment_metrics.py

```diff
@@ -153,26 +158,38 @@ def collect_metrics_for_enrollment(site, course_enrollment, course_sm, date_for=
     * else return the existing LearnerCourseGradesMetics record if it exists
     * else return None if no existing LearnerCourseGradesMetics record
     """
-    if not date_for:
-        date_for = datetime.utcnow().replace(tzinfo=utc).date()
 
     # The following are two different ways to avoide the dreaded error
     #     "Instance of 'list' has no 'order_by' member (no-member)"
     # See: https://github.com/PyCQA/pylint-django/issues/165
-    student_modules = course_sm.filter(
-        student_id=course_enrollment.user.id).order_by('-modified')
-    if student_modules:
-        most_recent_sm = student_modules[0]
-    else:
-        most_recent_sm = None
 
-    lcgm = LearnerCourseGradeMetrics.objects.filter(
+    if not student_modules:
+        student_modules = student_modules_for_course_enrollment(
+            site=site,
+            course_enrollment=course_enrollment).order_by('-modified')
```

Comment:
This is part of deciding which approach to take, so I implemented both

A. The caller has to provide the student module (SM) records for the learner. This is more efficient because the caller function only needs to call `collect_metrics_for_enrollment`. However, `collect_metrics_for_enrollment` must have the correct filter called (query only for the SM for this specific enrollment and reverse ordered by the modified date of the SM). It's a bit brittle

B. The caller does the check if any SM records for the enrollment, then does not pass this queryset to `collect_metrics_for_enrollment`. This is less efficient because two calls to query SM are made, but only when there are SM records 

At the core, if there are no SM records it means that the learner has not started any course activity = no progress made ever in the course, so we skip creating an LCGM record because it will contain no data

What I really need to do at some point is revisit the whole logic and streamline it. But not yet, not now as it will take time from working other more pressing needs